src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -301,136 +303,46 @@ class spell_mage_cone_of_cold : public SpellScriptLoader
-            {
-                uint8 level = GetHitUnit()->getLevel();
-                for (uint8 i = 0; i < MAX_CONJURE_REFRESHMENT_SPELLS; ++i)
-                {
-                    ConjureRefreshmentData const& spellData = _conjureData[i];
-                    if (level < spellData.minLevel || level > spellData.maxLevel)
-                        continue;
-                    GetHitUnit()->CastSpell(GetHitUnit(), spellData.spellId);
-                    break;
-                }
-            }
-
-            void Register() override
-            {
-                OnEffectHitTarget += SpellEffectFn(spell_mage_conjure_refreshment_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
-            }
-        };
-
-        SpellScript* GetSpellScript() const override
-        {
-            return new spell_mage_conjure_refreshment_SpellScript();
-        }
-};
-
-uint8 const MAX_CONJURE_REFRESHMENT_TABLE_SPELLS = 5;
-ConjureRefreshmentData const _conjureTableData[MAX_CONJURE_REFRESHMENT_TABLE_SPELLS] =
-{
-    { 73, 79, 120056 },
-    { 80, 84, 120055 },
-    { 85, 89, 120054 },
-    { 90, 99, 120053 },
-    { 100, 100, 167145 }
-};
-
-// 43987 - Conjure Refreshment Table
-class spell_mage_conjure_refreshment_table : public SpellScriptLoader
-{
 public:
-    spell_mage_conjure_refreshment_table() : SpellScriptLoader("spell_mage_conjure_refreshment_table") { }
+    spell_mage_conjure_refreshment() : SpellScriptLoader("spell_mage_conjure_refreshment") { }
 
-    class spell_mage_conjure_refreshment_table_SpellScript : public SpellScript
+    class spell_mage_conjure_refreshment_SpellScript : public SpellScript
     {
-        PrepareSpellScript(spell_mage_conjure_refreshment_table_SpellScript);
+        PrepareSpellScript(spell_mage_conjure_refreshment_SpellScript);
 
         bool Validate(SpellInfo const* /*spellInfo*/) override
         {
-            for (uint8 i = 0; i < MAX_CONJURE_REFRESHMENT_TABLE_SPELLS; ++i)
-                if (!sSpellMgr->GetSpellInfo(_conjureTableData[i].spellId))
-                    return false;
-            return true;
-        }
-
-        bool Load() override
-        {
-            if (GetCaster()->GetTypeId() != TYPEID_PLAYER)
+            if (!sSpellMgr->GetSpellInfo(SPELL_MAGE_CONJURE_REFRESHMENT) ||
+                !sSpellMgr->GetSpellInfo(SPELL_MAGE_CONJURE_REFRESHMENT_TABLE))
                 return false;
             return true;
         }
 
         void HandleDummy(SpellEffIndex /*effIndex*/)
         {
-            uint8 level = GetHitUnit()->getLevel();
-            for (uint8 i = 0; i < MAX_CONJURE_REFRESHMENT_TABLE_SPELLS; ++i)
-            {
-                ConjureRefreshmentData const& spellData = _conjureTableData[i];
-                if (level < spellData.minLevel || level > spellData.maxLevel)
-                    continue;
-                GetHitUnit()->CastSpell(GetHitUnit(), spellData.spellId);
-                break;
-            }
+            Unit* caster = GetCaster();
```

Comment:
Incorrect for `if (Group* group = caster->GetGroup())` I have warning for unused variable group